hypothesis/lms
# Remove `has_document_url` predicate

Remove the has_document_url predicate and use a conditional in the view

This refactor should be functionally equivalent but has a few advantages
that should be more obvious in follow up changes:

- We can remove the whole custom predicate infra as this is the only one
  that uses it

- We have more control over getting the document as that's part of
  "our code" vs being called by pyramid before picking a view. Based on
  this we'll be able to move document "getting" to an assignment look up.




## Testing

Sanity check with:

- Truncate assignments `docker compose exec postgres psql -U postgres -c "truncate assignment cascade"`
- As a student (`aunltd.S1` ) launch  https://aunltd.brightspacedemo.com/d2l/le/content/6782/viewContent/2431/View

you'll get the:


```
This assignment hasn't been configured yet. An instructor needs to launch the assignment to configure it.
```

message.


- Try with a teacher `HypothesisEng.Teacher`


https://aunltd.brightspacedemo.com/d2l/le/content/6782/viewContent/2431/View


you'll be able to configure it now and see the document.


- Launch it again https://aunltd.brightspacedemo.com/d2l/le/content/6782/viewContent/2431/View

you'll get the content again.
